noxfiles/git_nox.py

```diff
@@ -115,9 +122,23 @@ def next_release_increment(session: nox.Session, all_tags: List) -> Version:
         ),
         reverse=True,
     )
-    latest_release = releases[0]
+    latest_release = releases[0] if releases else None
     if not latest_release:  # this would be bad...
         session.error("Could not identify the latest release!")
+
+    if treat_rc_as_release:
+        rc_releases = sorted(  # sorted by Version - what we want!
+            (
+                Version(tag.name)
+                for tag in all_tags
+                if re.fullmatch(RC_TAG_REGEX, tag.name)
+            ),
+            reverse=True,
+        )
+        latest_rc_release = rc_releases[0] if rc_releases else None
```

Comment:
Is it checking for rc releases of only the latest version? In what case could it be `None`?

Reply:
hmm no actually this'll get all `rc` tags, i guess the `None` is there just as a safeguard to prevent an unhandled error. pretty sure the real reason is that i just c+p'd the function from another area here where it was possible that we'd not have a match 😄 

if you think it's problematic it'd probably be OK to remove and have an unhandled error, it's not like we should ever hit this if things are generally working fine

Reply:
gonna go ahead and merge as-is, lmk if you see a problem here though and i'm happy to address in a follow up 👍

Reply:
definitely not worth holding up the PR! As a future note though if there is a case that you really don't expect and could potentially signify an issue, something like `assert rc_releases is not None` is helpful documentation to other developers that that really is an unexpected state